packages/babel-traverse/src/scope/index.js

```diff
@@ -514,6 +515,8 @@ export default class Scope {
           if (local.identifier === id) continue;
 
           this.checkBlockScopedCollisions(local, kind, name, id);
+          // if no duplicates were found, set existing binding to null
+          local = null
```

Comment:
Seems like it's since `local` is passed to binding

https://github.com/babel/babel/blob/master/packages/babel-traverse/src/scope/binding.js#L30-L36

which sets `constantViolations`

https://github.com/babel/babel/blob/master/packages/babel-plugin-check-es2015-constants/src/index.js#L9-L11